### DISCUSSION TITLE
filesystem.readBinaryFile NE_RT_NATRTER

### DISCUSSION THREAD

**Discussion by jesuscc1993** [neutralinojs/neutralinojs#967] [Q&A] [answered]
I have the following code:

```
let imageBytes;
try {
  imageBytes = await Neutralino.filesystem.readBinaryFile({
    fileName: 'Z:/Txus/Images/Misc/Budgies/IMG_20210806_150530.jpg', // tried many other images and formats, but the error never changes
  });
} catch (err) {
  console.error(err);
}
```

And filesystem.readBinaryFile always throws the following error:
```
{
  code: 'NE_RT_NATRTER',
  message: "Native method execution error occurred. Failed because of: Unknown exception. Make sure that you've provided required parameters properly."
}
```

Based on the documentation I am pretty sure I have "provided required parameters properly", so what am I doing wrong?

**Comment by Blatman** [comment, marked as answer]
Ahh - the docs are a bit dodgy in places.

The target must be a string not an object as shown in the docs. The return will be a Uint8Array object.
eg:
var result = await Neutralino.filesystem.readBinaryFile(NL_PATH + '/trayIcon.png')
console.log('Result', result);   // this will be in the form of a Uint8Array

**Reply by jesuscc1993** [reply]
Ah, I put too much trust in them and didn't even try that.
It's working now; thank you very much.
